giskard/project.py

```diff
@@ -331,7 +331,7 @@ def _validate_classification_threshold_label(classification_labels, classificati
     @staticmethod
     def _validate_label_with_target(classification_labels, target_values=None):
         if target_values is not None:
-            if not isinstance(target_values, str):
+            if not all(isinstance(item, str) for item in target_values):
```

Comment:
Here's what I was talking about during a weekly meeting:

```python
from pandas.api.types import is_string_dtype
if not is_string_dtype(target_values):
...
```
Like this, you don't need to check each value individually. Also keep in mind that pandas `object` type can also store strings (which is handled by is_string_dtype)

Reply:
df.num.infer_objects()